### PR TITLE
GH-392: Improve automatically added import for React

### PR DESCRIPTION
**Pull request by qtran-n4**
Task: https://github.com/eclipse/n4js/issues/392

### CODE REVIEW

**Comment by qtran-n4** [comment]
@yoosiba Please review the changes. Thanks!

**Comment by yoosiba** [comment]
On this branch organise imports was is not working for projects without main module. Pls look into this (there are tests in the `org.eclipse.n4js.n4ide.spec.tests` that are failing). Please take care.

**Comment by qtran-n4** [comment]
I have fixed the failing tests. Turned out the fix should only be applied to module import.

**Comment by qtran-n4** [comment]
The builds are green!

File: plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/proposals/imports/FQNImporter.java [hunk 1]
```diff
@@ -78,6 +82,9 @@
 	@Inject
 	private ImportRewriter.Factory importRewriterFactory;
 
+	@Inject
+	IN4JSCore n4jsCore;
```
Comment (mmews-n4) [inline plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/proposals/imports/FQNImporter.java]: seems that this isn't used

File: plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/proposals/imports/ImportsAwareReferenceProposalCreator.java [hunk 2]
```diff
@@ -58,6 +60,9 @@
 	@Inject
 	private FQNImporter.Factory fqnImporterFactory;
 
+	@Inject
+	IN4JSCore n4jsCore;
```
Comment (mmews-n4) [inline plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/proposals/imports/ImportsAwareReferenceProposalCreator.java]: Can this be private?

Comment (qtran-n4) [reply plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/proposals/imports/ImportsAwareReferenceProposalCreator.java]: made private. done

**Comment by qtran-n4** [comment]
Thanks! Merging.

**Pull request merged**